colossalai/kernel/npu/mha/triangle_attn.py

```diff
@@ -0,0 +1,130 @@
+# coding=utf-8
+# Copyright (c) 2023, HUAWEI CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import torch
+from einops import rearrange
+
+HAS_NPU_TRIANGLE_ATTENTION = False
+try:
+    from torch_npu import npu_scaled_masked_softmax
```

Comment:
The condition is different from `get_attention_kernel()`

Reply:
NPUColoAttention will use spda attention by default. If it has npu_scaled_masked_softmax kernel, we will use triangle attention instrad.